src/mimeparser.rs

```diff
@@ -1084,6 +1085,28 @@ impl MimeMessage {
         }
     }
 
+    /// Remove unwanted, additional text parts.
+    /// Some mailinglist software add footers as separate mimeparts
+    /// eg. when the user-edited-content is html.
+    /// As these footers would appear as repeated, separate text-bubbles,
+    /// we remove them
+    fn maybe_remove_additional_text_parts(&mut self) {
+        if self.is_mailinglist_message() {
+            let text_part_cnt = self
+                .parts
+                .iter()
+                .filter(|p| p.typ == Viewtype::Text)
+                .count();
+            if text_part_cnt == 2 {
+                if let Some(last_part) = self.parts.last() {
+                    if last_part.typ == Viewtype::Text {
```

Comment:
What if there is an image attachment after two text parts? In this case there are two text parts, but you look at the last part, it's not text, so you don't delete the second text part?

Reply:
i want to catch only the footer for mailinglists with this pr - and that is always the last part, in all examples i've seen. 

i have not seen many mails with multiple text parts at all, however, maybe they exist, therefore also the explicit is_mailinglist() check.

Reply:
Then maybe rename it to something more specific, like `maybe_remove_inline_mailinglist_footer`.

Reply:
+1, i do a commit.